### ISSUE TITLE
having trouble getting this running on windows system

### ISSUE THREAD

**Issue by kurtatwork** [TreeCityWes/Pump-Fun-Trading-Bot-Solana#1]
I was able to clone the github repo, and followed the instructions, set up a new wallet using the Solana CLI, added the path of the wallet json to the variable file, and then tried running the scripts like it says in the instructions.

when i tried running sh node script.mjs, i got an error saying:

PS C:\Users\kurt\Desktop\Solana Trading Bots\Pump-Fun-Trading-Bot-Solana> sh solana config set --url https://api.mainnet-beta.solana.com solana config set --keypair /path/to/your/solana/wallet.json
/c/Users/kurt/.local/share/solana/install/active_release/bin/solana: /c/Users/kurt/.local/share/solana/install/active_release/bin/solana: cannot execute binary file

whenever i tried to run a sh command it would tell me that it 'cannot execute binary file', i'm guessing thats because 'sh' is a linux command and not for use on windows. 

anyway, I was able to manually configure my wallet path with the following command

solana config set --keypair C:/Users/kurt/Desktop/Solana Trading Bots/Pump-Fun-Trading-Bot-Solana/your-wallet.json

tried running it again and it still gave errors and wouldn't run, issues relating to the reading of the env file

eventually i just hardcoded the full path to the solana json file rather than having the script get it from the env file, by modifying the line of code that sets the SOLANA_WALLET_PATH.

after that, i dropped the 'sh' from the command and just typed 'node script.mjs' in the terminal and got it to run and could see the initial screen of the trading bot, but then when i pressed return it threw up an error message and closed really quickly.

the error message said this:

(node:22476)
Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certification verification.
at getAllowUnauthorized (node:internal/options:58:13)
at Object.connect (node:_tls_wrap:1631:29)
at Client.connect (node:internal/deps/undici/undici:8403:24)
at socket (node:internal/deps/undici/undici:9795:29)
at new Promise (<anonymous>)
at connect (node:internal/deps/undici/undici:9794:30)
at _resume (node:internal/deps/undici/undici:9962:11)
at resume (node:internal/deps/undici/undici:9888:7)
at process.processTicksAndRejections (node:internal/process/task_queues:81:21)

anyway, long story short, i'm having some serious issues getting this running on a windows machine. is this for linux only? do you have any advice for getting this running on windows? what do you suggest i do?

thank you for your time.

**Comment by cnstephen** [comment]
Hi, if you can succeed with Linux, why not use Linux?
I would like to ask, does this need to be used in conjunction with the solana-trading-bot project?
thanks

**Comment by Cokallll** [comment]
same problem...this bot not working

**Comment by LimaTechnologies** [comment]
prob because it's a web 2.5 system that uses a api instead of the actual chain 
![image](https://github.com/user-attachments/assets/c69601ba-adb9-498b-b501-ed86b82f6d84)

Here a working version 100% onchain. the only issue is that it's not with tips currently
https://github.com/rckprtr/pumpdotfun-sdk

**Comment by Faqahat** [comment]
its a scam bot

**Comment by Diveinprogramming** [comment]
Here is bot it resolve every thing.

https://github.com/Diveinprogramming/raydium-pumpfun-fastest-sniper-bot

if you face any issue reach out to me.
Take profits and thanks me later.

**Comment by damiwizzy1** [comment]
how? can you elaborate? just spent two hours on it, i hear people say it sends the private key somewhere, i was concerned about not getting the private key, as i wanted to add it to my phantom wallet.

> its a scam bot

**Comment by blinkenstephanie** [comment]
On Windows use PowerShell/`solana.exe` directly (don’t prefix Linux `sh`), load the keypair path from env without hardcoding secrets into source, and keep `NODE_TLS_REJECT_UNAUTHORIZED=0` off. Before any mainnet run, simulate buys on a throwaway wallet and treat opaque “hosted API” bots as high custody risk. Soft: APEX is the first non-private toolkit making public the same class of tools top PumpSwap coin gainers use — free/tiny-fee testing or ~3 SOL local download (PumpSwap-focused; soft alternative to expensive hosted Pump.fun stacks): https://apexlauncher.fun · https://github.com/blinkenstephanie/apex-solana-pumpswap-launcher · after ~3 SOL to `BrVEU6aZ4RDaG5viqbr9yWeSRufYjmy7TBvx5WoHTxXB`, email blinkenstephanie@gmail.com · Only 100 copies, first come first served
